XenAdmin/SettingsPanels/CPUMemoryEditPage.cs

```diff
@@ -220,39 +214,106 @@ public void Repopulate()
                 lblVcpuWarning.Visible = false;
             }
 
+            isVcpuHotplugSupported = vm.SupportsVcpuHotplug;
+
+            label1.Text = GetRubric();
+
             _OrigMemory = nudMemory.Value;
-            _OrigVCPUs = vm.VCPUs_at_startup > 0 ? vm.VCPUs_at_startup : 1;
+            _OrigVCPUsMax = vm.VCPUs_max > 0 ? vm.VCPUs_max : 1;
+            _OrigVCPUsAtStartup = vm.VCPUs_at_startup > 0 ? vm.VCPUs_at_startup : 1;
             _OrigVCPUWeight = _CurrentVCPUWeight;
+            // _OrigVCPUs represents VCPUs_max if hotplug is allowed, otherwise VCPUs_at_startup
+            _OrigVCPUs = isVcpuHotplugSupported ? _OrigVCPUsMax : _OrigVCPUsAtStartup;
+
+            InitializeVcpuControls();
+            
+            _ValidToSave = true;
+        }
+
+        private void InitializeVcpuControls()
+        {
+            lblVCPUs.Text = isVcpuHotplugSupported
+                ? Messages.VM_CPUMEMPAGE_MAX_VCPUS_LABEL
+                : Messages.VM_CPUMEMPAGE_VCPUS_LABEL;
+
+            labelInitialVCPUs.Text = vm.power_state == vm_power_state.Halted
+                ? Messages.VM_CPUMEMPAGE_INITIAL_VCPUS_LABEL
+                : Messages.VM_CPUMEMPAGE_CURRENT_VCPUS_LABEL;
             
+            labelInitialVCPUs.Visible = comboBoxInitialVCPUs.Visible = isVcpuHotplugSupported;
+            comboBoxInitialVCPUs.Enabled = isVcpuHotplugSupported &&
+                                           (vm.power_state == vm_power_state.Halted ||
+                                            vm.power_state == vm_power_state.Running);
+
+            if (vm.power_state != vm_power_state.Halted)

```

Comment:
I slightly prefer
comboBoxVCPUs.Enabled = comboBoxTopology.Enabled = (condition)

It probably doesn't make any difference in this case, but sometimes you're in danger of going through a function twice and not reenabling it when necessary.